src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTask.java

```diff
@@ -146,8 +146,9 @@ public String version() {
 
   private void checkMapping(String index, SinkRecord record) {
     if (!config.shouldIgnoreSchema(record.topic()) && !existingMappings.contains(index)) {
-      if (!client.hasMapping(index)) {
-        client.createMapping(index, record.valueSchema());
+      String targetIndex = client.resolveIndexIfAlias(index);
```

Comment:
There might be an issue w.r.t how the mappings are being caches here (in `existingMappings`). If the alias write index is changed then this will not create the mapping in the new write index as we are caching the name of the alias in that case (end of this method). 

Adding `targetIndex` to the cache is not that great either as it means the checkMapping will result in a call to the current write index on each put. 

Also, in hindsight there can be a race condition between a write index update for an alias and a mapping update here i.e. by the time we write the data using an alias the backing write index might have changes. Hence we might end up writing a document to an index without a mapping at that point.

Reply:
Yes, I see, my goal here was to support the async write to index with a name different from the topic name - basically a static alias. 
Since full support is not possible as there is no efficient way to create mappings in case of full alias usage (would need to check the current write index with Elasticsearch on every write), I would suggest disabling that feature in case of alias usage. If the topic name points the connector to an alias, it needs to manage underlying indices and mappings manually. What do you think?